File: src/main/java/io/vproxy/vfx/ui/pane/ClickableFusionPane.java
package io.vproxy.vfx.ui.pane;

import io.vproxy.vfx.theme.Theme;
import javafx.event.EventHandler;
import javafx.scene.Cursor;
import javafx.scene.paint.Color;

public class ClickableFusionPane extends FusionPane {
    private EventHandler<?> handler;

    public ClickableFusionPane() {
        super();
        getNode().setCursor(Cursor.HAND);
    }

    public ClickableFusionPane(boolean manuallyHandleOuterRegion) {
        super(manuallyHandleOuterRegion);
        getNode().setCursor(Cursor.HAND);
    }

    @Override
    protected AbstractFusionPane buildRootNode() {
        return new FusionPaneImpl() {
            @Override
            protected Color downColor() {
                return Theme.current().fusionButtonDownBackgroundColor();
            }

            @Override
            protected void onMouseClicked() {
                var h = handler;
                if (h != null) {
                    h.handle(null);
                }
            }
        };
    }

    public void setOnAction(EventHandler<?> handler) {
        this.handler = handler;
    }
}

File: src-ts/src/ui/pane/ClickableFusionPane.ts
import { Theme } from '../../theme/Theme.js';
import { Color } from '../../javafx/color.js';
import type { VMouseEvent } from '../../javafx/Node.js';
import { FusionPane } from './FusionPane.js';
import { FusionPaneImpl } from './FusionPane.js';

export type ClickableFusionPaneHandler = (e: VMouseEvent | null) => void;

export class ClickableFusionPane extends FusionPane {
  private handler: ClickableFusionPaneHandler | null = null;

  constructor();
  constructor(manuallyHandleOuterRegion: boolean);
  constructor(manuallyHandleOuterRegion?: boolean) {
    super(manuallyHandleOuterRegion ?? true);
    this.getNode().setCursor('pointer');
  }

  protected override buildRootNode() {
    const outer = this;
    return new (class extends FusionPaneImpl {
      protected override downColor(): Color {
        return Theme.current().fusionButtonDownBackgroundColor();
      }

      protected override onMouseClicked(): void {
        const h = outer.handler;
        if (h !== null) {
          h(null);
        }
      }
    })(outer);
  }

  setOnAction(handler: ClickableFusionPaneHandler | null): void {
    this.handler = handler;
  }
}
